pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -2555,80 +2562,56 @@ func TestContextInvalid(t *testing.T) {
 		tasks: []PipelineTask{{
 			Name:    "bar",
 			TaskRef: &TaskRef{Name: "bar-task"},
-			Params: []Param{{
-				Name: "a-param", Value: ParamValue{StringVal: "$(context.pipeline.missing)"},
-			}},
```

Comment:
don't remove this test

same applies to the other test cases

Reply:
Sorry, this was an outstanding comment from a previous PR to isolate the functionality of context variables in the matrix since it was testing both valid context variables in matrix and valid context variables in params.

 I have added an additional set of tests TestContextValidMatrix() and TestContextInvalidMatrix() to isolate the context variables in the matrix by removing Task.Params and kept the existing TestContextValid() and TestContextInalid()